src/main/java/com/fasterxml/jackson/core/json/async/NonBlockingUtf8JsonParserBase.java

```diff
@@ -467,7 +467,7 @@ private final JsonToken _finishBOM(int bytesHandled) throws IOException
         }
         _pending32 = bytesHandled;
         _minorState = MINOR_ROOT_BOM;
-        return (_currToken = JsonToken.NOT_AVAILABLE);
+        return _updateToken(JsonToken.NOT_AVAILABLE);
```

Comment:
This would work incorrectly since it can happen multiple times in a row; should only update once token type is determined.
So maybe third method `_updateTokenNA()` or such, similar to `_updateTokenToNull()`?